controllers/helmchart_controller.go

```diff
@@ -463,6 +464,9 @@ func (r *HelmChartReconciler) buildFromHelmRepository(ctx context.Context, obj *
 		tlsConfig *tls.Config
 		loginOpts []helmreg.LoginOption
 	)
+	// Used to login with the repository declared provider
+	ctxTimeout, cancel := context.WithTimeout(ctx, repo.Spec.Timeout.Duration)
```

Comment:
Due to this usage, can you also update the description of spec.timeout in https://github.com/fluxcd/source-controller/blob/2010eef3747c91790508e5138c0dd145c7bc21aa/api/v1beta2/helmrepository_types.go#L71 ?